EIPS/eip-223.md

```diff
@@ -0,0 +1,389 @@
+2. This token is more gas-efficient when depositing tokens to contracts.
+3. This token allows for `_data` recording for financial transfers.
+
+
+## Specification
+
+Token
+Contracts that works with tokens
+
+### Methods
+
+NOTE: An important point is that contract developers must implement `tokenReceived` if they want their contracts to work with the specified tokens.
+
+If the receiver does not implement the `tokenReceived` function, consider the contract is not designed to work with tokens, then the transaction must fail and no tokens will be transferred. An analogy with an Ether transaction that is failing when trying to send Ether to a contract that did not implement `receive() payable`.
+
+
+#### totalSupply
+
+```js
+
+function totalSupply() constant returns (uint256 totalSupply)
+
+```
+
+Get the total token supply
+
+#### name
+
+```js
+
+function name() constant returns (string _name)
+
+```
+
+Get the name of token
+
+#### symbol
+
+```js
+
+function symbol() constant returns (bytes32 _symbol)
+
+```
+
+Get the symbol of token
+
+#### decimals
+
+```js
+
+function decimals() constant returns (uint8 _decimals)
+
+```
+
+Get decimals of token
+
+#### standard
+
+```js
+
+function standard() constant returns (string _standard)
+
+```
+
+Get the standard of token contract. For some services it is important to know how to treat this particular token. If token supports [ERC-223](./eip-223.md) standard then it must explicitly tell that it does.
+
+This function **MUST** return "ERC-223" for this token standard. If no "standard()" function is implemented in the contract then the contract must be considered to be ERC-20.
+
+#### balanceOf
+
+```js
+
+function balanceOf(address _owner) constant returns (uint256 balance)
+
+```
+
+Get the account balance of another account with address _owner
+
+
+#### transfer(address, uint)
+
+```js
+
+function transfer(address _to, uint _value) returns (bool)
+
+```
+
+Needed due to backwards compatibility reasons because of ERC-20 transfer function doesn't have `bytes` parameter. This function must transfer tokens and invoke the function `tokenReceived(address, uint256, bytes calldata)` in `_to`, if _to is a contract. If the `tokenReceived` function is not implemented in `_to` (receiver contract), then the transaction must fail and the transfer of tokens should be reverted.
+
+#### transfer(address, uint, bytes)
+
+```js
+
+function transfer(address _to, uint _value, bytes calldata _data) returns (bool)
+
+```
+
+function that is always called when someone wants to transfer tokens.
+This function must transfer tokens and invoke the function `tokenReceived (address, uint256, bytes)` in `_to`, if _to is a contract. If the `tokenReceived` function is not implemented in `_to` (receiver contract), then the transaction must fail and the transfer of tokens should not occur. 
```

Comment:
```suggestion
This function must transfer tokens and invoke the function `tokenReceived (address, uint256, bytes)` in `_to`, if `_to` is a contract. If the `tokenReceived` function is not implemented in `_to` (receiver contract), then the function call must fail and the transfer of tokens should not occur. 
```

Reply:
I would say that this was a correct wording because the whole transaction must fail on this error. If there are multiple calls within one transaction then it must fail altogether, not just this function call.